oauth2_http/java/com/google/auth/oauth2/ServiceAccountCredentials.java

```diff
@@ -127,6 +128,28 @@ public ServiceAccountCredentials(
   public ServiceAccountCredentials(
       String clientId, String clientEmail, PrivateKey privateKey, String privateKeyId,
       Collection<String> scopes, HttpTransportFactory transportFactory, URI tokenServerUri) {
+    this(clientId, clientEmail, privateKey, privateKeyId, scopes, transportFactory, tokenServerUri, null);
+  }
+
+  /**
+   * Constructor with minimum identifying information and custom HTTP transport.
+   *
+   * @param clientId Client ID of the service account from the console. May be null.
+   * @param clientEmail Client email address of the service account from the console.
+   * @param privateKey RSA private key object for the service account.
+   * @param privateKeyId Private key identifier for the service account. May be null.
+   * @param scopes Scope strings for the APIs to be called. May be null or an empty collection,
+   *        which results in a credential that must have createScoped called before use.
+   * @param transportFactory HTTP transport factory, creates the transport used to get access
+   *        tokens.
+   * @param tokenServerUri URI of the end point that provides tokens.
+   * @param serviceAccountUser Email of the user account to impersonate, if delegating domain-wide
+   *        authority to the service account.
+   */
+  public ServiceAccountCredentials(
+      String clientId, String clientEmail, PrivateKey privateKey, String privateKeyId,
+      Collection<String> scopes, HttpTransportFactory transportFactory, URI tokenServerUri,
+      String serviceAccountUser) {
```

Comment:
I think I want to avoid an explosion of constructors. We are stuck-ish with the existing ones, but maybe make your new one private because you really just want to use it from `fromPkcs8`. Although, we seem to also have an explosion of methods with that name, too. Maybe we could even hide that one - someone could use the existing `fromPkcs8` function and call `createDelegated` on the result to get what they want.